rpc/core/bridge_commitments.go

```diff
@@ -0,0 +1,232 @@
+		LastResultsMerkleProof:      txMerkleProof,
+	}, nil
+}
+
+// fetchBridgeCommitmentLeaves takes an end exclusive range of heights and fetches its
+// corresponding BridgeCommitmentLeafs.
+func (env *Environment) fetchBridgeCommitmentLeaves(start, end uint64) ([]ctypes.BridgeCommitmentLeaf, error) {
+
+	bridgeCommitmentLeaves := make([]ctypes.BridgeCommitmentLeaf, 0, end-start)
+	for height := start; height < end; height++ {
+
+		currentBlock := env.BlockStore.LoadBlock(int64(height))
+		if currentBlock == nil {
+			return nil, fmt.Errorf("couldn't load block %d", height)
+		}
+
+		bridgeCommitmentLeaves = append(bridgeCommitmentLeaves, ctypes.BridgeCommitmentLeaf{
+			Height:          height,
+			LastResultsHash: currentBlock.Header.LastResultsHash,
+		})
+	}
+
+	return bridgeCommitmentLeaves, nil
+}
+
+// encodeBridgeCommitment takes a height and a last result hash, and returns the equivalent of
+// `abi.encode(...)` in Ethereum. To match `abi.encode(...)`, the height is padding to 32 bytes.
+func (env *Environment) encodeBridgeCommitment(leaves []ctypes.BridgeCommitmentLeaf) ([][]byte, error) {
+
+	encodedLeaves := make([][]byte, 0, len(leaves))
+	for _, leaf := range leaves {
+
+		// Pad to match `abi.encode` on Ethereum.
+		paddedHeight, err := To32PaddedHexBytes(leaf.Height)
+		if err != nil {
+			return nil, err
+		}
+
+		encodedLeaf := append(paddedHeight, leaf.LastResultsHash...)
+		encodedLeaves = append(encodedLeaves, encodedLeaf)
+	}
+
+	return encodedLeaves, nil
+}
+
+// To32PaddedHexBytes takes a number and returns its hex representation padded to 32 bytes.
+// Used to mimic the result of `abi.encode(number)` in Ethereum.
+func To32PaddedHexBytes(number uint64) ([]byte, error) {
+	hexRepresentation := strconv.FormatUint(number, 16)
+	// Make sure hex representation has even length.
+	// The `strconv.FormatUint` can return odd length hex encodings.
+	// For example, `strconv.FormatUint(10, 16)` returns `a`.
+	// Thus, we need to pad it.
+	if len(hexRepresentation)%2 == 1 {
+		hexRepresentation = "0" + hexRepresentation
+	}
+	hexBytes, hexErr := hex.DecodeString(hexRepresentation)
+	if hexErr != nil {
+		return nil, hexErr
+	}
+	paddedBytes, padErr := padBytes(hexBytes, 32)
+	if padErr != nil {
+		return nil, padErr
+	}
+	return paddedBytes, nil
+}
+
+// padBytes Pad bytes to given length.
+func padBytes(byt []byte, length int) ([]byte, error) {
+	l := len(byt)
+	if l > length {
+		return nil, fmt.Errorf(
+			"cannot pad bytes because length of bytes array: %d is greater than given length: %d",
+			l,
+			length,
+		)
+	}
+	if l == length {
+		return byt, nil
+	}
+	tmp := make([]byte, length)
+	copy(tmp[length-l:], byt)
+	return tmp, nil
+}
+
+// validateBridgeCommitmentRange runs basic checks on the asc sorted list of
+// heights that will be used successively to generate bridge commitments over
+// the defined set of height/s.
+func (env *Environment) validateBridgeCommitmentRange(start, end uint64) error {
```

Comment:
Where's the asc sorted list of height? I think we should reword to "height range"

Reply:
WDYT?

```
// validateBridgeCommitmentRange runs basic checks on the range of heights
// that will be used to generate bridge commitments from successive blocks.
```

Reply:
ok